### DISCUSSION TITLE
Display freezes completely

### DISCUSSION THREAD

**Discussion by Dopfen** [tbnobody/OpenDTU#2009] [Q&A]
### What happened?

Hi, 
I'm using OpenDTU on a display unit with OLED display. 
In irregular periods, the display freezes completely. It happens surely once within 4 days always in the morning, some minutes after the micronverter had started and some data was displayed. 

It first occurred when using firmware 23.12.19, offering the great new diagram drawing. It happens never before. 
Then, I almost installed every sucessing firmware. Even with the nevest 24.5.6 the issue occurrs.
 
Some info:
Microinverter is Hoymiles HM-400.

Display unit:
ESP32-D0WD-V3 with OLED display. 

Firmware:
Version: 24.5.6
Host name: OpenDTU-28ED64
SDK versiom: 4.4.6
Configuration version: 0.1.28

See also screenshots from memory and heap. 

Notes:
For OLED displays, freezing pixels tend to burn into the display, so it is bad for the hardware. 

When the display freezed, it is still possible to enter the OpenDTU dashboard.

Would be great if you could help me or at least investigate the issue.

Best regards,
Daniel

### To Reproduce Bug

Not predictable. 

### Expected Behavior

No freezing display. 

### Install Method

Pre-Compiled binary from GitHub

### What git-hash/version of OpenDTU?

24.5.6

### Relevant log/trace output

```Shell
No log file yet.
If helpful, I could upload a log file (please show me how)
```

### Anything else?

See pics from the device and screenshots from OpenDTU dashboard for detailed info on memory and heap. 

![DSC_1701](https://github.com/tbnobody/OpenDTU/assets/170804583/6c76a8f0-1191-4320-9f9a-69f5e84eb9b8)
![Screenshot_20240521-195435](https://github.com/tbnobody/OpenDTU/assets/170804583/e315e908-84ad-4684-bf76-5a2956ee8457)
![Screenshot_20240521-195404](https://github.com/tbnobody/OpenDTU/assets/170804583/5ed4386d-1bc4-4766-9e04-8001021351f3)

**Comment by tbnobody** [comment]
That lookls like that either the display is defekt or the power supply is bad. (Power Supply itself, wiring, capacitor etc.)
When updating more pixels (like redrawing a whole diagram) the display consumes more power.

**Comment by tbnobody** [comment]
I am going to convert this to a discussion as it is for sure no software issue.

**Comment by Dopfen** [comment]
Hi @tbnobody,
Thanks for the quick reply.

I didn't mention that the picture of the display unit look strange, as it is caused by photographic flicker. The picture quality of the display is perfect.

Nevertheless, it is important to know that it is not a software issue and could be caused by a insufficient power supply. 
My power supply gives 5V, 2.1A, which should be enough. The provider said that 2A is required. 
-> I'll test different power supplies the next days.

**Comment by barnosch** [comment]
Guess i have the same issue. 

If i repower the OpenDTU, the display is working again. 
I can reach the Webpage everytime, so it is no complete freeze.

**Reply by ghost** [reply]
Hello,
I agree with @tbnobody, the display is defect.
This happens occasionally, see here; #1906

**Comment by Dopfen** [comment]
Hi,
Update after some days of testing a more powerful power supply (from JBL Charge 3), which gives 5V, 2.2A and has better quality as the initial one producing the freeze:

While the initial one lets the display freeze after max. 4 days, it took up to 9 days until this happend with the JBL supply. Sury not perfect, but an improvement. 
Currently, I'm testing a second time. 

So, tbnobody's suggestion to check the power supply was very helpful.

The thread, Schneer mentioned, is so deep hardware technical for me in order to check if the display defect or connectors are bad. 
Anyway, I'll contact the person I got the display unit from. 

Thanks so far and greetings, 
Daniel
